ocs_ci/ocs/ui/base_ui.py

```diff
@@ -1485,20 +1500,1136 @@ def __init__(self):
+
+    def nav_back_main_topology_view(self, soft: bool = False):
+        """
+        Navigate back to the higher hierarchy level in the topology view. Can be used in any level of topology,
+        but will be performed only if button back exists when param soft is True
+
+        Args:
+            soft (bool, optional): If True, performs a navigation back if the "back" button is present.
+                                   If False, performs a navigation back by clicking the "back" button.
+                                   Defaults to False.
+        Note:
+            The "back" button exists only on the deployment level.
+        """
+        if soft and len(self.get_elements(self.topology_loc["back_btn"])):
+            logger.info("navigate topology view to higher hierarchy")
+            self.do_click(self.topology_loc["back_btn"])
+        elif not soft:
+            self.do_click(self.topology_loc["back_btn"])
+
+    def check_entity_selected(self, entity_name):
+        """
+        Checks if the specified entity is selected.
+
+        Args:
+            entity_name (str): Name of the entity.
+
+        Returns:
+            bool: True if the entity is selected, False otherwise.
+
+        """
+        from ocs_ci.ocs.ui.helpers_ui import format_locator
+
+        # selected node should contain pf-m-selected in the class name
+        attribute = self.get_element_attribute(
+            format_locator(
+                self.topology_loc["entity_box_select_indicator"], entity_name
+            ),
+            "class",
+            True,
+        )
+        if not attribute:
+            return
+        else:
+            return "pf-m-selected" in attribute
+
+    def select_entity_with_search_bar(self, entity_name):
+        """
+        Selects the specified entity using the search bar.
+
+        Args:
+            entity_name (str): Name of the entity to select.
+
+        """
+        logger.info(f"selecting '{entity_name}' with search bar")
+        self.do_send_keys(self.topology_loc["topology_search_bar"], entity_name)
+        self.do_click(self.topology_loc["topology_search_bar_enter_arrow"])
+
+    def reset_search_bar(self):
+        """
+        Resets the search bar.
+        """
+        logger.info("reset search")
+        self.do_click(self.topology_loc["topology_search_bar_reset_search"])
+
+    def is_entity_present(self, entity_name) -> bool:
+        """
+        Checks if the specified entity is present.
+
+        Args:
+            entity_name (str): The name of the entity to search for.
+
+        Returns:
+            bool: True if the entity is present, False otherwise.
+        """
+
+        from ocs_ci.ocs.ui.helpers_ui import format_locator
+
+        return bool(
+            self.get_elements(
+                format_locator(self.topology_loc["select_entity"], entity_name)
+            )
+        )
+
+
+class TopologyTab(DataFoundationDefaultTab, AbstractTopologyView):
```

Comment:
Can you move this class to anther file?

Reply:
yes, I will do it, as I mentioned in the comment to PR after this PR. I cannot separate only this class, this is a chain of classes needed to be distributed in their specific files. The problem why I can not do it with TopologyTab only, because it requires import of dependencies, and thus creates circular import problem - https://github.com/red-hat-storage/ocs-ci/issues/7704